### PR TITLE
nrf: Split the ble module into a shared part and the port implementation

### PR DESCRIPTION
**Pull request by arturo182**
This allows other ports to implement these shared bindings.

This is one part of #995

### CODE REVIEW

**Comment by dhalbert** [comment]
This is very clean. I am thinking that maybe we want to make a class that lives under `ble`, like `ble.Bluetooth`, `ble.Device`, or `ble.BLE`. Better suggestions welcomed. Then we have a singleton instance that is the device. We can then use properties, which we can't do with a regular module. So

```
if ble.bluetooth.enabled:
    ...
ble.bluetooth.enabled = False
print(ble.bluetooth.address)
```
etc.

Later there could be other things that live under `ble`, maybe message classes?? I don't know enough now to make good suggestions. Maybe `ble` should be `bleio`??

Some other examples of this kind of singleton class in CircuitPython are `microcontroller.cpu` (instance of `microcontroller.Processor`) and `nvm.Bytearray`.

**Comment by arturo182** [comment]
Yes, I wanted to use a property for enabled and address but had two hesitations: 1. would need a class, 2. would break the current API.

Regarding having a class, I think best name would be `Adapter`, pretty common terminology in the BT world. We could keep all other classes like `Central`, `Peripheral`, `Service`, `Characteristic` inside a `bleio` module as well, but that would break compatibility with `bluepy`.

**Comment by arturo182** [comment]
Let me know if you want me to move the functions to a Adapter singleton.

**Comment by tannewt** [comment]
I vote to use a singleton. 

We'll break compatibility with upstream due to our use of properties anyway. We'll break with `bluepy` because we need more APIs and their methods are non-standard camel case.

So, lets do what we think is best. Thanks!

**Comment by arturo182** [comment]
Woo, let the Bluetooth overhaul begin! Expect many PRs soon :D

**Comment by arturo182** [comment]
So we agree on `bleio`?

**Comment by tannewt** [comment]
Ya, `bleio` works for me. *io is kinda lame but its consistent with our previous stuff and helps make it a unique name.

**Comment by arturo182** [comment]
@tannewt @dhalbert Updated :)

File: shared-bindings/bleio/Adapter.c [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+//|
+//|         MAC address of the BLE adapter. (read-only)
+//|
+
+#define BLE_ADDRESS_LEN 17
+
+STATIC mp_obj_t bleio_adapter_get_enabled(mp_obj_t self) {
+    const bool enabled = common_hal_bleio_adapter_get_enabled();
+
+    return enabled ? mp_const_true : mp_const_false;
```
Comment (dhalbert) [inline shared-bindings/bleio/Adapter.c]: You can do `return mp_obj_new_bool(common_hal_bleio_adapter_get_enabled());`. This is common in other `shared-bindings` implementations.

File: ports/nrf/mpconfigport.h [hunk 2]
```diff
@@ -126,27 +126,27 @@
...
-#ifndef MICROPY_PY_UBLUEPY
-#define MICROPY_PY_UBLUEPY                       (0)
+#ifndef MICROPY_PY_BLEIO
+#define MICROPY_PY_BLEIO                        (0)
 #endif
 
-#ifndef MICROPY_PY_BLE_NUS
-#define MICROPY_PY_BLE_NUS                       (0)
+#ifndef MICROPY_PY_UBLUEPY
+#define MICROPY_PY_UBLUEPY                      (0)
```
Comment (dhalbert) [inline ports/nrf/mpconfigport.h]: Q: are we going to get rid of `ubluepy` at some point or does it serve some purpose in other nrf ports?

Comment (arturo182) [reply ports/nrf/mpconfigport.h]: Yes, next step is port `ubluepy` classes to `bleio`, I just like to keep things separated and do them one step at a time.

File: shared-bindings/bleio/Adapter.c (hunk 1 above)
Comment (arturo182) [reply shared-bindings/bleio/Adapter.c]: Didn't know that one, will fix!

File: ports/nrf/mpconfigport.h (hunk 2 above)
Comment (dhalbert) [reply ports/nrf/mpconfigport.h]: OK, sounds good!

**Comment by arturo182** [comment]
@dhalbert Fixed!

**Pull request merged**